samples/bluetooth/ipsp/README.rst

```diff
@@ -19,22 +19,164 @@ Testing with a Linux host
+BlueZ package) and its ``scan on`` command:
+
+.. code-block:: console
+
+   $ bluetoothctl
+   [NEW] Controller A3:24:97:EB:D6:23 ubuntu-0 [default]
+   [NEW] Device D7:5C:D6:18:14:87 Zephyr
+   [NEW] Device E1:E7:F9:56:EC:06 Zephyr
+   [NEW] Device C8:12:C5:08:86:E1 Zephyr
+   [bluetooth]# scan on
+   Discovery started
+   [NEW] Device DC:98:FB:22:CA:3A Zephyr
+
+When started, ``bluetoothctl`` shows all BLE (and likely, BT/EDR) devices it
+knows about. As discussed above, the IPSP uses static random addresses, so
+entries like above accumulate and become stale. You need to be extra careful
+to find an entry for the active address. The best approach may be to reset
+your board after issuing ``scan on`` command, this way it will reinitialize
+with the BLE address which will be discovered after the command.
+
+As an alternative to ``bluetoothctl``, you can use legacy ``hcitool`` utility
+which talks directly to hardware and always shows fresh scan results:
+
+.. code-block:: console
+
+   $ sudo hcitool lescan
+   LE Scan ...
+   CB:AF:14:57:D8:6E (unknown)
+   CB:AF:14:57:D8:6E Test IPSP node
+
+After you have found the board's BLE address, connect to the board (as root):
+
+.. code-block:: console
+
+   # echo "connect <bdaddr> <type>" > /sys/kernel/debug/bluetooth/6lowpan_control
+
+Where ``<bdaddr>`` is the BLE address and ``<type>`` is BLE address type:
+1 for public address and 2 for static random address. As you can see from
+the IPSP sample output above, it uses a static random address. So, with the
+sample output above, the command will be:
+
+.. code-block:: console
+
+   # echo "connect CB:AF:14:57:D8:6E 2" > /sys/kernel/debug/bluetooth/6lowpan_control
+
+Once connected a dedicated interface will be created, usually bt0. You can verify this
+with the following command:
+
+.. code-block:: console
+
+   # ifconfig
+   bt0       Link encap:UNSPEC  HWaddr F8-2F-A8-FF-FE-EB-6D-8C-00-00-00-00-00-00-00-00
+             inet6 addr: fe80::fa2f:a8ff:feeb:6d8c/64 Scope:Link
+             UP POINTOPOINT RUNNING MULTICAST  MTU:1280  Metric:1
+             RX packets:2 errors:0 dropped:3 overruns:0 frame:0
+             TX packets:6 errors:0 dropped:0 overruns:0 carrier:0
+             collisions:0 txqueuelen:1000
+             RX bytes:92 (92.0 B)  TX bytes:233 (233.0 B)
+
+As can be seen from the output, only a link-local IPv6 address was assigned
+to the interface.
+
+At this point, you can test IPv6 connectivity (and discover your board's IPv6
+address) by pinging "All local-link nodes" IPv6 address:
+
+.. code-block:: console
+
+   # ping6 -I bt0 ff02::1
+   PING ff02::1(ff02::1) from fe80::fa54:a8ff:feeb:218f bt0: 56 data bytes
+   64 bytes from fe80::fa54:a8ff:feeb:218f: icmp_seq=1 ttl=64 time=0.088 ms
+   64 bytes from fe80::c9af:14ff:fe57:d86e: icmp_seq=1 ttl=64 time=285 ms (DUP!)
+
+For each ping packet, both your host and the BLE board send a reply. You
+can see the board's reply marked as `(DUP!)`. You can ping the board
```

Comment:
sorry, should be with double backticks:

     ``(DUP!)``

Reply:
Fixed